lighthouse-core/test/config/config-test.js

```diff
@@ -33,6 +34,20 @@ describe('Config', () => {
     assert.notEqual(config, newConfig);
   });
 
+  it('doesn\'t change directly injected plugins', () => {
```

Comment:
may also want to add a test to `getGathererClass` on `GatherRunner` that it correctly exposes and validates a live plugin